src/config.ts

```diff
@@ -52,13 +52,14 @@ export interface GenerationConfig {
   shift_fill_factor?: number;
   repetition_penalty?: number;
   // Shared by MLC and OpenAI APIs
-  top_p?: number;
-  temperature?: number;
+  top_p?: number | null;
```

Comment:
What would be the default top_p and temperature if they are not provided in the ChatCompletion request?

Reply:
There are actually no default values -- if not provided, we use the value in `ChatConfig`, which is read in from `mlc-chat-config.json` in `ChatModule.reload()` (which is potentially overridden by `ChatOptions`). It is required that all fields are present in `mlc-chat-config.json`. I agree that this is not very straightforward and added some comments to the docstring.

Reply:
It is arguable whether we want to keep these fields in `mlc-chat-config.json` (top_p, temperature, max_gen_len, etc.) or we should make them an argument during runtime, so that we have one less source to consolidate with. Currently we have three sources:
- `mlc-chat-config.json` (represented as a `ChatConfig`, read during `reload()`)
- ChatOptions (partial `ChatConfig`, optionally specified during `reload()`)
- GenerationConfig (optional, only affects a single generation -- hence fields like conv template cannot be specified)

But this is something to discuss/coordinate with MLC as well

Reply:
Or perhaps we do not need to coordinate with mlc-llm and can simply not use `mlc-chat-config.json` in webLLM. If we can have universal default values, we do not depend on `mlc-chat-config.json` at all -- all other fields are read in from the model library's metadata, unlike mlc-llm.

Looking at the fields in `ChatConfig`, `tokenizer_files` can be replaced by simply searching within the huggingface weight repo; all fields below `mean_gen_len` can have a universal (i.e. for all models) default value. Whether we need to read `conv_template` from `mlc-chat-config.json` depends on what we decide on the conversation template enhancement.

```
  // First three fields affect the entire conversation, i.e. used in `ChatModule.reload()`
  tokenizer_files: Array<string>;
  conv_config?: Partial<ConvTemplateConfig>;
  conv_template: string;
  // Fields below can be swapped per-generation via `GenerationConfig`
  // Fields only used in MLC
  mean_gen_len: number;
  max_gen_len: number;
  shift_fill_factor: number;
  repetition_penalty: number;
  // Fields shared by MLC and OpenAI APIs
  top_p: number;
  temperature: number;
```